packages/block-library/src/breadcrumbs/index.php

```diff
@@ -448,21 +448,69 @@ function block_core_breadcrumbs_get_terms_breadcrumbs( $post_id, $post_type ) {
 		return array();
 	}
 
-	// Find the first taxonomy that has terms assigned to this post.
+	/**
+	 * Filters the preferred taxonomy and term for breadcrumbs on a per-post-type basis.
+	 *
+	 * Allows developers to specify which taxonomy and term should be used in breadcrumbs
+	 * when a post has multiple taxonomies or terms assigned.
+	 *
+	 * The preferred taxonomy must have terms assigned to the post, otherwise falls back to the first available taxonomy.
+	 *
+	 * The preferred term must exist and be assigned to the post. If the post has only one term, it's used regardless of preference.
+	 * If the preferred term is not found, falls back to the first term.
+	 *
+	 * @since 7.0.0
+	 *
+	 * @param array  $preferences {
+	 *     Array of breadcrumb preferences. Default empty array.
+	 *
+	 *     @type string $preferred_taxonomy Optional. Taxonomy slug to prefer for breadcrumbs.
+	 *     @type string $preferred_term     Optional. Term slug to prefer when post has multiple terms.
+	 * }
+	 * @param string $post_type   The post type slug.
+	 */
+	$preferences = apply_filters( 'block_core_breadcrumbs_post_type_preferences', array(), $post_type );
+
 	$taxonomy_name = null;
 	$terms         = array();
-	foreach ( $taxonomies as $taxonomy ) {
-		$post_terms = get_the_terms( $post_id, $taxonomy->name );
-		if ( ! empty( $post_terms ) && ! is_wp_error( $post_terms ) ) {
-			$taxonomy_name = $taxonomy->name;
-			$terms         = $post_terms;
-			break;
+
+	// Try preferred taxonomy first if specified.
+	if ( ! empty( $preferences['preferred_taxonomy'] ) ) {
+		$preferred_taxonomy = wp_list_filter( $taxonomies, array( 'name' => $preferences['preferred_taxonomy'] ) );
+		if ( ! empty( $preferred_taxonomy ) ) {
+			$preferred_taxonomy = reset( $preferred_taxonomy );
```

Comment:
Nits:

The same variable $preferred_taxonomy is reassigned to a different semantic type (first a list of matches, then a match). I say semantic type because technically they are always arrays. We should avoid this.
The mechanics of running a filter to then extract the first result feel a little strained, but no big deal.